src/HotChocolate/Core/src/Execution/DependencyInjection/InternalServiceCollectionExtensions.cs

```diff
@@ -166,7 +166,7 @@ internal static IServiceCollection TryAddDefaultCaches(
     {
         services.TryAddSingleton<IDocumentCache>(
             _ => new DefaultDocumentCache());
-        services.TryAddSingleton<IPreparedOperationCache>(
+        services.TryAddTransient<IPreparedOperationCache>(
```

Comment:
I've made the global operation cache transient, so we can keep the same configuration API via `.AddOperationCache(capacity: 123)`, but each executor can create a new (transient) instance for its own schema services.

Reply:
It's a bit unfortunate that you'd end up with a different instance, if you were to resolve the `IPreparedOperationCache` through the regular service collection...

Do you have any better ideas to still retain the ability to register a custom implementation or should we just get rid of that capability anyways?

Reply:
Yes... I think we split configuration options an instance.